onto-viewer-core/src/main/java/org/edmcouncil/spec/ontoviewer/core/ontology/data/handler/classes/ClassHandler.java

```diff
@@ -0,0 +1,187 @@
+    this.stringIdentifier = stringIdentifier;
+    this.axiomsHandler = axiomsHandler;
+    this.usageExtractor = usageExtractor;
+    this.taxonomyExtractor = taxonomyExtractor;
+    this.extractSubAndSuper = extractSubAndSuper;
+    this.qnameHandler = qnameHandler;
+    this.particularAnnotationPropertyHandler = particularAnnotationPropertyHandler;
+    this.individualDataHelper = individualDataHelper;
+    this.inheritedAxiomsHandler = inheritedAxiomsHandler;
+  }
+
+  public OwlListDetails handleParticularClass(OWLClass owlClass) {
+    var configurationData = applicationConfigurationService.getConfigurationData();
+
+    var ontology = ontologyManager.getOntology();
+    var classIri = owlClass.getIRI();
+    var resultDetails = new OwlListDetails();
+
+    try {
+      resultDetails.setLabel(labelProvider.getLabelOrDefaultFragment(owlClass));
+
+      OwlDetailsProperties<PropertyValue> axioms = axiomsHandler.handleAxioms(owlClass, ontology);
+      List<PropertyValue> subclasses = extractSubAndSuper.getSubclasses(axioms);
+      List<PropertyValue> subclasses2 = extractSubAndSuper.getSubclasses(owlClass);
+      List<PropertyValue> taxElements2 = taxonomyExtractor.extractTaxonomyElements(subclasses2);
+
+      OwlDetailsProperties<PropertyValue> directSubclasses = extractSubAndSuper.handleDirectSubclasses(
+          owlClass);
+      OwlDetailsProperties<PropertyValue> individuals = new OwlDetailsProperties<>();
+      if (configurationData.getToolkitConfig().isIndividualsEnabled()) {
+        individuals = individualDataHelper.handleInstances(ontology, owlClass);
+      }
+
+      OwlDetailsProperties<PropertyValue> usage = new OwlDetailsProperties<>();
+      if (configurationData.getToolkitConfig().isUsageEnabled()) {
+        usage = usageExtractor.extractUsageForClasses(owlClass, ontology);
+      }
+
+      OwlDetailsProperties<PropertyValue> inheritedAxioms =
+          inheritedAxiomsHandler.handleInheritedAxioms(ontology, owlClass);
+
+      OntologyGraph ontologyGraph = new OntologyGraph(0);
+      if (configurationData.getToolkitConfig().isOntologyGraphEnabled()) {
+        // 'Nothing' has all restrictions, we don't want to display that.
+        if (!owlClass.getIRI().equals(OWLRDFVocabulary.OWL_NOTHING.getIRI())) {
+          ontologyGraph = graphDataHandler.handleGraph(owlClass, ontology, 0, 0);
+        }
+      }
+      subclasses = extractSubAndSuper.filterSubclasses(subclasses);
+
+      OwlTaxonomyImpl taxonomy = taxonomyExtractor.extractTaxonomy(taxElements2, owlClass.getIRI(),
+          ontology,
+          AXIOM_CLASS);
+      taxonomy.sort();
+
+      OwlDetailsProperties<PropertyValue> annotations =
+          particularAnnotationPropertyHandler.handleAnnotations(owlClass.getIRI(), ontology,
+              resultDetails);
+
+      var copyright = copyrightHandler.getCopyright(classIri);
+      var license = licenseHandler.getLicense(classIri);
+      var qname = qnameHandler.getQName(classIri);
```

Comment:
it can be reduced into one line code lines with `resultDetails.addAllProperties()` under this